cases/models.py

```
@@ -41,15 +51,34 @@ def validate_url_list(value):
+            validator(stripped_link)
+            item["link"] = stripped_link
+
+            role = item.get("role")
+            if role is not None:
+                try:
+                    SourceLinkRole(role)
+                except ValueError:
+                    raise ValidationError(
+                        f"Invalid role '{role}'. Must be one of "
+                        f"{[r.value for r in SourceLinkRole]}."
+                    )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)